platform/linux-generic/odp_schedule_scalable.c

```diff
@@ -950,18 +950,19 @@ static int _schedule(odp_queue_t *from, odp_event_t ev[], int num_evts)
 	for (i = 0, first = NULL; i < ts->num_schedq; i++, first = NULL) {
 		sched_queue_t *schedq = ts->schedq_list[i];
 		sched_elem_t *elem;
+		time_stamp = _odp_cpu_global_time();
 restart_same:
 		elem = schedq_peek(schedq);
 		if (odp_unlikely(elem == NULL)) {
 			/* Schedq empty, look at next one. */
 			continue;
 		}
 		if (is_pktin(elem)) {
-			/* Pktio ingress queue */
-			if (first == NULL)
-				first = elem;
-			else if (elem == first) /* Wrapped around */
+			if (elem->loop_check[odp_cpu_id()] == time_stamp) { /* Wrapped around */
 				continue; /* Go to next schedq */
+			} else {
+				elem->loop_check[odp_cpu_id()] = time_stamp;
```

Comment:
`odp_thread_id()` should be used here. Its range is always from 0 to odp_thread_count_max() - 1.